apps/wallet/src/generated/station/station.did

```diff
@@ -605,7 +605,63 @@ type ChangeExternalCanisterOperation = record {
   arg_checksum : opt Sha256Hash;
 };
 
-type CreateExternalCanisterOperationInput = record {};
+type CreateExternalCanisterOperationInput = record {
+  // By default, the canister id is created by the system.
+  //
+  // If provided, the canister must exist.
+  existing_canister_id : opt principal;
+  // The initial cycles to allocate to the canister.
+  //
+  // Ignored if the canister already exists.
+  initial_cycles : opt nat64;
```

Comment:
What do you think about an `operation : Operation` variant 

AddExisting { canister_id : principal } 
CreateNew { initial_cycles : nat64 }

That way you can't represent conflicting input eg. providing both `existing_canister_id` and `initial_cycles`, it would need less documentation, and it would be extendable.

Reply:
was meaning to add comments to [/api/spec.did](https://github.com/dfinity/orbit/pull/305/files#diff-55b7ab3f382de0fb84e61ee21b84337109b8dc6e2e0a82ebb70f8440c9806c99)

Reply:
I like the approach you suggested, wdyt of something like the following:

```
type CreateExternalCanisterOperationKindCreateNew = record {
  // The initial cycles to allocate to the canister.
  //
  // If not set, only the minimal amount of cycles required to create the
  // canister will be allocated.
  initial_cycles : opt nat64;
};

type CreateExternalCanisterOperationKindAddExisting = record {
  // The canister id to use.
  canister_id : principal;
};

type CreateExternalCanisterOperationKind = variant {
  // A new canister is created.
  CreateNew : CreateExternalCanisterOperationKindCreateNew;
  // An existing canister is added to the station.
  AddExisting : CreateExternalCanisterOperationKindAddExisting;
};

type CreateExternalCanisterOperationInput = record {
  // The kind of create operation to perform.
  kind : CreateExternalCanisterOperationKind;
  // The name of the external canister.
  name : text;
  // The description of the external canister.
  description : opt text;
  // The labels of the external canister.
  labels : opt vec text;
  // What operations are allowed on the canister.
  permissions : ExternalCanisterPermissionsInput;
  // The request policies for the canister.
  request_policies : ExternalCanisterRequestPoliciesInput;
};
  ```